zulipterminal/ui_tools/boxes.py

```diff
@@ -232,12 +232,30 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                 return key
             header = self.contents[0][0]
             # toggle focus position
-            if self.focus_position == 0 and self.to_write_box is None:
-                if header.focus_col == 0:
-                    header.focus_col = 1
-                    return key
+            if self.focus_position == 0:
+                if self.to_write_box is None:
+                    if header.focus_col == 0:
+                        stream_name = header[0].edit_text
+                        if not self.model.is_valid_stream(stream_name):
+                            self.view.set_footer_text("Invalid stream name", 3)
+                            return key
+
+                        header.focus_col = 1
+                        return key
+                    else:
+                        header.focus_col = 0
                 else:
-                    header.focus_col = 0
+                    recipient_box = header.original_widget
+                    recipient_emails = recipient_box.edit_text.split(', ')
```

Comment:
This works for some cases, but assumes precise separation in this way - eg it fails for 'email1 , email2', which is unexpected (and the feedback isn't clear?)

That said, we don't currently really validate what gets passed to the API calls, and could likely use ids instead of emails in future based on a general migration to use ids there in the server.

Reply:
@neiljp  I can fix the issue with the spaces, but what is the expected feedback?

Reply:
The current situation is twofold:
* that it complains about 'email1 , email2', where I would expect it to be fine - I should have indicated that both are 'valid' in my thought-experiment/test-case. That is, I'd expect 'email1,email2', 'email1, email2' and 'email1 , email2' to be valid? That said, I'm not sure what the server accepts in terms of whitespace of this kind
* the message produced is also a problem. eg. 'chat2406@chat.zulip.org ' also fails but shows what looks like a valid address in the error message, similar for the cases above without `, ` between emails.

Reply:
The server ignores whitespaces completely so it doesn't matter where it is. I fixed the code to follow the same, so this should not be a problem anymore.